src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -148,285 +149,124 @@ public static DateTime GetLastAccessTimeUtc(string path)
             return File.GetLastAccessTimeUtc(path);
         }
 
-        // Returns an array of filenames in the DirectoryInfo specified by path
-        public static string[] GetFiles(string path)
-        {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
-
-            return InternalGetFiles(path, "*", SearchOption.TopDirectoryOnly);
-        }
-
-        // Returns an array of Files in the current DirectoryInfo matching the 
-        // given search pattern (i.e. "*.txt").
-        public static string[] GetFiles(string path, string searchPattern)
-        {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
-            if (searchPattern == null)
-                throw new ArgumentNullException(nameof(searchPattern));
+        public static string[] GetFiles(string path) => GetFiles(path, "*", enumerationOptions: EnumerationOptions.Compatible);
 
-            return InternalGetFiles(path, searchPattern, SearchOption.TopDirectoryOnly);
-        }
+        public static string[] GetFiles(string path, string searchPattern) => GetFiles(path, searchPattern, enumerationOptions: EnumerationOptions.Compatible);
 
-        // Returns an array of Files in the current DirectoryInfo matching the 
-        // given search pattern (i.e. "*.txt") and search option
         public static string[] GetFiles(string path, string searchPattern, SearchOption searchOption)
+            => GetFiles(path, searchPattern, EnumerationOptions.FromSearchOption(searchOption));
+
+        public static string[] GetFiles(string path, string searchPattern, EnumerationOptions enumerationOptions)
         {
             if (path == null)
```

Comment:
all the callers could move these two null checks inside InternalEnumeratePaths

Reply:
Was trying to avoid the extra check when I knew I was providing good data. I'll move the path one.

Reply:
Never mind, moving them both.